core/src/main/scala/org/http4s/rho/AuthedContext.scala

```diff
@@ -57,13 +48,13 @@ class AuthedContext[F[_]: Monad, U] extends FailureResponseOps[F] {
     }
   }
 
-  /* Get the authInfo object from request. */
-  def getAuth(req: Request[F]): U = {
-    req.attributes.get[U](authKey).get
-  }
+  /** Get the authInfo object from request */
+  def getAuth(req: Request[F]): Option[U] =
+    req.attributes.get(authKey)
```

Comment:
Good catch, `get` should almost never be called on an `Option`, however, there is a more fundamental problem here that is beyond the scope of this PR to fix.   You should not be able to call `getAuth` if your `RhoService` hasn't been wrapped in a `AuthMiddleware` but the current design allows this.  So, for now, I propose we just make the error message you get for calling `getAuth` better, and file an issue to correct our auth framework so it doesn't allow this.

Thinking replacing `get` with `getOrElse(throw new RuntimeException("Misconfiguration: RhoService called getAuth but has not had AuthMiddleware applied"))`.  Certainly open to other exception messages, but looking for something more descriptive than `get` would throw alone.

Reply:
Raised it [here](https://github.com/http4s/rho/issues/256). For me `None` makes much more sense than `RuntimeException` because design of `AuthMiddleware` implies that it can return `None`.